csharp/src/SeedLang.Shell/Program.cs

```diff
@@ -13,20 +13,83 @@
 // limitations under the License.
 
 using System;
+using CommandLine;
+using CommandLine.Text;
 using SeedLang.Common;
 
 namespace SeedLang.Shell {
   class Program {
-    static void Main(string[] _) {
+    class Options {
+      [Option('r', "rule", Required = false, Default = "statement",
+        HelpText = "The parse rule of the source code." +
+                   " Allowed values are identifier, number, string, expression or statement.")]
+      public string Rule { get; set; }
+
+      [Option('l', "language", Required = false, Default = "python",
+        HelpText =
+          "The language of the source code. Allowed values are python, lua or javascript.")]
+      public string Language { get; set; }
+
+      [Option('t', "type", Required = false, Default = "source",
+        HelpText = "Run type. Allowed values are dryrun, source or bytecode.")]
+      public string Type { get; set; }
+    }
+
+    static void Main(string[] args) {
+      var parserResult = Parser.Default.ParseArguments<Options>(args);
+      var helpText = HelpText.AutoBuild<Options>(parserResult, h => h, e => e);
+      parserResult.WithParsed(options => {
+        Console.Error.WriteLine(helpText.Heading);
+        Console.Error.WriteLine(helpText.Copyright);
+        Console.Error.WriteLine();
+        RunOptions(options);
+      });
+    }
+
+    private static void RunOptions(Options options) {
       // TODO: add a REPL class to encapsulate the input and execution process.
+      Language language = Language.Python;
+      ParseRule rule = ParseRule.Statement;
+      switch (options.Rule) {
+        case "identifier":

```

Comment:
Consider case enum.GetName(ParseRule.Identifier).ToLower() ?

Here  and the rest.

Reply:
The case of the switch statement must be constant. I added a TODO here to consider using dictionary later.